pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -499,6 +501,8 @@ func (l *LibvirtDomainManager) SetGuestTime(vmi *v1.VirtualMachineInstance) erro
 	return nil
 }
 
+var loopbackAddress = ip.GetLoopbackAddress()
```

Comment:
Global alert, global alert!
First, please locate global declarations at the top of the file, it took me a while to detect it.

Second, you need to really reason on why you use a global here, especially that this will run pre-main even which sounds really bad to me.

Reply:
This is the method that Vladik suggested to use, 
it appears on more places in the manager.go official code
We need this because Interface mocking is for RPC only according Vladik,
and the unit tests which doesn't have a cluster fails,
I will investigate, it might be that a delay would be enough, or increasing timeouts in the unit tests.

Reply:
Debugging i see the following error
https://travis-ci.org/github/kubevirt/kubevirt/builds/690593931#L884

{"component":"","level":"error","msg":"failed to create unix socket for proxy service","pos":"migration-proxy.go:334","reason":"listen tcp [::1]:49152: bind: cannot assign requested address","timestamp":"2020-05-24T12:37:17.008418Z"}

The function when running "make goveralls"
runs on the travis Virtual machine which doesnt support IPv6, but according the above error
the VM have ipv6_disable=0 in the sysfs.

We should decide on a way to stub this function for the unit tests which doesnt have a real cluster.
I would suggest to stick with the current method, either a variable or a function, as this is the common method
on production code, and was suggested by Vladik as well.
We see its a must to stub it out.
Vladik asked that it wont be a function if possible (which what i done already),
so lets please talk tommorow with him,
the 4 of us, or decide one direction and unblock this PR.
We can change it, if needed, in a following PR.
If its fine with you, i will do rest of the nits, so we are ready with lgtm for Vladik's approval.

Thanks

Reply:
Unit tests cannot run functions on the OS (including reading from sysfs).
It needs to be replaced in the test through a mock/stub.

AFAIK, doing that requires function pointers on a struct that represents the parts that needs replacements.
The mocking mechanics is one way to do this, but it requires an interfaces and does a bit more than is needed IMO. I would just do it manually here instead of that automatic mocking.

Reply:
I did it manually here, this is what you mean that you suggest ?
Interface as DomainManager is for RPC as Vladik explained.

Reply:
Github is swallowing such threads when code is changed and this thread merits a reset anyway.
I'm moving the discussion to the new location where the global was moved.